source/air/src/context.rs

```diff
@@ -33,9 +35,11 @@ pub(crate) struct AxiomInfo {
 pub enum ValidityResult {
     Valid,
     Invalid(Model, Error),
+    SingularInvalid(Error),
```

Comment:
Again, I don't like special-casing, as we'll need to handle them one by one.

Is there a reason why this cannot reuse `Invalid`? Maybe with an optional `Model` on `Invalid`?

Reply:
Sure, I will try with `Option<Model>`